### PR TITLE
Add OSS Notice issuance function to Self-Check

### PR DESCRIPTION
**Pull request by namkyu1999**
## Description

Add OSS Notice issuance function to Self-Check like below.

Untitled

Untitled (1)

Untitled (2)

## What to do later
- Making an error appear when the OSS Notice button is clicked before the save button is pressed --> finished
- Added SPDX issuance function

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: db/initdb.d/fosslight_create.sql [hunk 1]
```diff
@@ -1449,13 +1482,15 @@ DELETE FROM `PRE_OSS_COMPONENTS`;
 -- 테이블 fosslight.PRE_OSS_COMPONENTS_LICENSE 구조 내보내기
 DROP TABLE IF EXISTS `PRE_OSS_COMPONENTS_LICENSE`;
 CREATE TABLE IF NOT EXISTS `PRE_OSS_COMPONENTS_LICENSE` (
+  `COMPONENT_ID2` int(11) NOT NULL AUTO_INCREMENT,
```
Comment (riyenas0925) [inline db/initdb.d/fosslight_create.sql:1485]: What does COMPONENT_ID2 mean? What's the difference with COMPONENT_ID?

File: src/main/webapp/WEB-INF/views/admin/selfCheck/edit.jsp [hunk 2]
```diff
@@ -141,9 +141,9 @@
 		<div class="btnLayout">
 			<input id="delete" type="button" value="Delete" class="btnColor left selfCheckDelete" /><!-- 2018-07-19 choye 추가 class에  selfCheckDelete -->
 			<span class="right">
+				<input type="button" value="OSS Notice" onclick="createTabInFrame('Notice','#/selfCheck/verification/'+'${project.prjId}')" class="btnColor red btnNotice srcBtn"/>
```
Comment (riyenas0925) [inline src/main/webapp/WEB-INF/views/admin/selfCheck/edit.jsp:144]: I think you need to increase the button width

image

**Comment by riyenas0925** [comment]
@namkyu1999 
I need a query to update the current db. (Because I can't delete the old db and recreate it)

File: db/initdb.d/fosslight_create.sql (hunk 1 above)
Comment (namkyu1999) [reply db/initdb.d/fosslight_create.sql:1485]: @riyenas0925 

Currently, the COMPONENT_ID of self-check is a combination of (REFERENCE_ID + REFERENCE_DIV + COMPONENET_IDX). You can query PRE_OSS_COMPONENTS_LICENSE in multiple mapper functions only if you have a unique id that is AUTO_INCREMENT. So I added COMPONENT_ID2(Same as COMPONENT_ID in OSS_COMPONENTS_LICENSE).

File: src/main/webapp/WEB-INF/views/admin/selfCheck/edit.jsp (hunk 2 above)
Comment (namkyu1999) [reply src/main/webapp/WEB-INF/views/admin/selfCheck/edit.jsp:144]: Fix it like below!
image

Comment (riyenas0925) [reply src/main/webapp/WEB-INF/views/admin/selfCheck/edit.jsp:144]: LGTM :+1:

**Comment by namkyu1999** [comment]
A commit that allows you to enter copyright text in Self-Check has been added, and pr has been updated.

image

**Comment by riyenas0925** [comment]
@namkyu1999 
Please clean up your commits.

**Comment by namkyu1999** [comment]
> @namkyu1999 Please clean up your commits. (It contains unnecessary changes.)

Everything is reflected!

**Comment by soimkim** [comment]
@namkyu1999 , 
For users running FOSSLight, please leave a DB change query as a comment.

**Comment by namkyu1999** [comment]
@soimkim,
I'm not sure what that means.
Please tell me a little bit more about where and how to comment.

**Comment by soimkim** [comment]
Need more reivew.

**Comment by soimkim** [comment]
@namkyu1999 , I hid the OSS Notice button of Self-check.
It is scheduled to change to visible after detailed modifications are reflected.

**Pull request merged**